src/persist/src/mem.rs

```diff
@@ -69,8 +71,25 @@ impl MemBlobCore {
         Ok(())
     }
 
-    fn list_keys(&self) -> Result<Vec<String>, ExternalError> {
-        Ok(self.dataz.keys().cloned().collect())
+    fn list_keys_and_metadata(
+        &self,
+        key_prefix: Option<&str>,
+        f: &mut (dyn FnMut(BlobMetadata) + Send + Sync),
+    ) -> Result<(), ExternalError> {
+        for (key, value) in &self.dataz {
+            if let Some(prefix) = key_prefix {
+                if !key.starts_with(prefix) {
+                    continue;
+                }
+            }
+
+            f(BlobMetadata {
+                key: &key,
+                size_in_bytes: value.len() as u64,
```

Comment:
`u64::cast_from(usize)` (which may require importing mz_ore::cast::CastFrom). mz uses this to declare in a single place that we only run on 64-bit systems